File: domain/src/commonMain/kotlin/com/voxenlabs/domain/server/usecases/GetStoredServersUseCase.kt
package com.voxenlabs.domain.server.usecases

import com.voxenlabs.domain.server.models.Server
import com.voxenlabs.domain.server.repositories.ServerRepositoryInterface
import org.koin.core.annotation.Factory

@Factory
class GetStoredServersUseCase(
    val serverRepository: ServerRepositoryInterface,
) {
    suspend operator fun invoke(): List<Server> = serverRepository.getStoredServers()
}

File: src/modules/Voxen.Client.Domain/ServerDefinitions/UseCases/GetStoredServersUseCase.cs
using Voxen.Client.Domain.ServerDefinitions.Models;
using Voxen.Client.Domain.ServerDefinitions.RepositoryInterfaces;

namespace Voxen.Client.Domain.ServerDefinitions.UseCases
{
    public class GetStoredServersUseCase(IServerRepository serverRepository)
    {
        private readonly IServerRepository serverRepository = serverRepository;

        public List<Server> Invoke()
        {
            return serverRepository.GetStoredServers();
        }
    }
}
